src/Core/GeoPoint.php

```diff
@@ -0,0 +1,183 @@
+<?php
+/**
+ * Copyright 2016 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Cloud\Core;
+
+use InvalidArgumentException;
+
+/**
+ * Represents a geographical point.
+ *
+ * Example:
+ * ```
+ * use Google\Cloud\Core\GeoPoint;
+ *
+ * $point = new GeoPoint(37.423147, -122.085015);
+ * ```
+ *
+ * @see https://cloud.google.com/datastore/reference/rest/Shared.Types/LatLng LatLng
+ */
+class GeoPoint
+{
+    /**
+     * @var float
+     */
+    private $latitude;
+
+    /**
+     * @var float
+     */
+    private $longitude;
+
+    /**
+     * Create a GeoPoint.
+     *
+     * Ints will be converted to floats. Values not passing the `is_numeric()`
+     * check will result in an exception.
+     *
+     * @param float|int $latitude
+     * @param float|int $longitude
+     * @throws InvalidArgumentException
+     */
+    public function __construct($latitude, $longitude)
+    {
+        $this->setLatitude($latitude);
+        $this->setLongitude($longitude);
+    }
+
+    /**
+     * Get the latitude
+     *
+     * Example:
+     * ```
+     * $latitude = $point->latitude();
+     * ```
+     *
+     * @return float
+     */
+    public function latitude()
+    {
+        $this->checkContext('latitude', func_get_args());
+        return $this->latitude;
+    }
+
+    /**
+     * Set the latitude
+     *
+     * Non-numeric values will result in an exception
+     *
+     * Example:
+     * ```
+     * $point->setLatitude(42.279594);
+     * ```
+     *
+     * @param int|float $latitude The new latitude
+     * @return GeoPoint
+     * @throws InvalidArgumentException
+     */
+    public function setLatitude($latitude)
+    {
+        if (is_numeric($latitude)) {
```

Comment:
This class is shared with Datastore, which is GA. Removing the setters would require a BC break. Therefore, if keeping the setters for Firestore is unacceptable, I'll create a Firestore-specific Lat/Lng class. Which would you prefer?

Reply:
As per @lukesneeringer there are some efforts on the way to separate out the different clients and their resources into product-specific repos (such as php-firestore). If that happens before we publicly announce this client, I would prefer if we remove the setters. Otherwise, being able to share this code probably outweighs the risk of API divergence across Firestore.

Reply:
Thanks @schmidt-sebastian. We already split each service into a separate package. In our case, it would be `google/cloud-firestore`. This class is part of the Core package ([`google/cloud-core`](https://github.com/googlecloudplatform/google-cloud-core)). It is required as a dependency for every other package, and thus is unaffected by any split as far out on the roadmap as I'm aware of.

Based on that, and your direction, I'll leave it as is unless I hear otherwise.